### DISCUSSION TITLE
Pico 4/Neo 3 Passthrough Modes & Improvements/Fixes

### DISCUSSION THREAD

**Discussion by korejan** [korejan/ALVR#72] [Announcements]
The latest nightly builds of [ALXR](https://github.com/korejan/ALXR-nightly/releases/latest) now has support for [passthrough modes](https://youtu.be/DMe-xes5B1w), for Pico 4 & Neo 3. There have also been improvements & fixes for Pico headsets:

- 2x Pico build types
    - `alxr-client-pico-[-debug].apk` - This is is based on the latest Pico OpenXR SDK (2.1.0) for PUI versions 5.2.x and up
    - `alxr-client-pico-v4[-debug].apk` - This build is based on the previous Pico OpenXR SDK for PUI versions 4.7.x - 5.1.x
-  Passthrough Modes - support for both build types
    - Usage is Left Menu (- button) + A (blend mode) **or** B (mask mode)
- Fixes head/controller pose lag during fast motions.
- Better default resolutions for ALVR relative 100% video resolution

Hand-tracking is supported but currently disabled due to [bugs ](https://developer-global.pico-interactive.com/community/post/7185104130313420806) in Pico's OpenXR runtime. Pico have responded, acknowledged the issues and are working on fixes.

Pico Neo 3 (Link) users, PUI 5.3.x will work best as there are fixes/improvements such as fixing haptic feedback for OpenXR apps (ALXR in this case).

More pico specific improvements to come, mask mode to be reworked for better results and have options.

**Comment by suppressio** [comment]
Hi,

I'm a Pico 4 user and I would thank you for this release! 
This new version is needed to continue using Pico 4 (PUI 5.3.x) streaming on Linux.

I would like to share my user experience, which may be useful to someone else.
I upgraded my Pico 4 to PUI 5.3.2 and could no longer get the ALXR client (the "old" nightly version) to work properly.

The application starts but, after about 20 seconds of playing, the whole system of my Pico becomes unstable (the main UI crashes, it is no longer possible to return to "home", you can only force a reboot).

Now I installed `alxr-client-pico-v4.apk` and everything works perfectly.
So thanks again!

**Reply by korejan** [reply]
Hi, thanks, so the latest version that uses Pico OpenXR SDK 2.1.0 (the one without -v4) was causing issues too? I don't have a Pico 4 but I have had people test it with PUI 5.3.2.

**Reply by suppressio** [reply]
I'll try it out and let you know asap...

**Reply by suppressio** [reply]
Also `alxr-client-pico.apk` is ok, it's working fine. 
I then tried again with an older release (the one produced by commit [9d0989d](https://github.com/korejan/ALVR/commit/9d0989d7b39fd2068d1bfb05cda4af0314da356f) ) and I confirm the issues.

**Comment by KandZ123** [comment]
Thank you for this release.
I have a question about Mask Mode (key color range masking).
The key color is only black?

**Reply by korejan** [reply]
Currently the mask mode is black with small threshold so not strictly pure black. The mask mode shader will be re-written in the near the future with a better implementation and options for user defined chroma/luma keys.
